src/services/rfqm_service.ts

```diff
@@ -513,6 +528,184 @@ export class RfqmService {
         return;
     }
 
+    /**
+     * Submit a transaction, re-submit if necessary, and poll for resolution
+     */
+    public async completeSubmissionLifecycleAsync(
+        orderHash: string,
+        workerAddress: string,
+        callData: string,
+    ): Promise<void> {
+        let submissionsMap: SubmissionsMap = {};
+
+        let gasPrice: BigNumber;
+        let nonce: number;
+        let gasEstimate: number;
+
+        [gasPrice, nonce, gasEstimate] = await Promise.all([
+            this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync(),
+            this._blockchainUtils.getNonceAsync(workerAddress),
+            this._blockchainUtils.estimateGasForExchangeProxyCallAsync(callData, workerAddress),
+        ]);
+
+        const expectedTakerTokenFillAmount = this._blockchainUtils.getTakerTokenFillAmountFromMetaTxCallData(callData);
+
+        const firstSubmission = await this._submitTransactionAsync(
+            orderHash,
+            workerAddress,
+            callData,
+            gasPrice,
+            nonce,
+            gasEstimate,
+            expectedTakerTokenFillAmount,
+        );
+        submissionsMap[firstSubmission.transactionHash!] = firstSubmission;
+
+        let isTxMined = false;
+        let isTxFinalized = false;
+        while (!isTxFinalized) {
+            await delay(TRANSACTION_WATCHER_SLEEP_TIME_MS);
+
+            const statusCheckResult = await this._checkSubmissionMapReceiptsAndUpdateDbAsync(
+                submissionsMap,
+                expectedTakerTokenFillAmount,
+            );
+            isTxMined = statusCheckResult.isTxMined;
+            isTxFinalized = statusCheckResult.isTxFinalized;
+            submissionsMap = statusCheckResult.submissionsMap;
+
+            if (!isTxMined) {
+                const newGasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
+
+                if (gasPrice.lt(newGasPrice)) {
+                    gasPrice = newGasPrice;
+                    const submission = await this._submitTransactionAsync(
+                        orderHash,
+                        workerAddress,
+                        callData,
+                        gasPrice,
+                        nonce,
+                        gasEstimate,
+                        expectedTakerTokenFillAmount,
+                    );
+                    submissionsMap[submission.transactionHash!] = submission;
+                }
+            }
+        }
+    }
+
+    /**
+     * Check for receipts from the tx's in a SubmissionsMap object
+     * Update database with status of all tx's
+     */
+    private async _checkSubmissionMapReceiptsAndUpdateDbAsync(
+        submissionsMap: SubmissionsMap,
+        expectedTakerTokenFillAmount: BigNumber,
+    ): Promise<SubmissionsMapStatus> {
+        let isTxMined: boolean = false;
+        let isTxFinalized: boolean = false;
```

Comment:
super nit: do not need the explicit `boolean` type declaration as type inference will determine that